contracts/drafts/ERC777/ERC777.sol

```diff
@@ -0,0 +1,402 @@
+pragma solidity ^0.5.2;
+
+import "./IERC777.sol";
+import "./IERC777Recipient.sol";
+import "./IERC777Sender.sol";
+import "../../math/SafeMath.sol";
+import "../../utils/Address.sol";
+import "../IERC1820Registry.sol";
+
+/**
+ * @title ERC777 token implementation
+ * @author etsvigun <utgarda@gmail.com>, Bertrand Masius <github@catageeks.tk>
+ */
+contract ERC777 is IERC777 {
+    using SafeMath for uint256;
+    using Address for address;
+
+    IERC1820Registry private _erc1820 = IERC1820Registry(0x1820b744B33945482C17Dc37218C01D858EBc714);
+
+    string private _name;
+
+    string private _symbol;
+
+    mapping(address => uint256) private _balances;
+
+    uint256 private _totalSupply;
+
+    uint256 private _granularity;
+
+    bytes32 constant private TOKENS_SENDER_INTEFACE_HASH = keccak256("ERC777TokensSender");
```

Comment:
Small typo: `TOKENS_SENDER_INTEFACE_HASH` should be `TOKENS_SENDER_INTERFACE_HASH`